File: 0938-Range-Sum-of-BST/0938-Range-Sum-of-BST.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rangeSumBST(self, root: Optional[TreeNode], low: int, high: int) -> int:
        return self.dfs(root, low, high, 0)
    def dfs(self, root, low, high,sum):
        if not root:
            return sum
        if low <= root.val <= high:
            sum += root.val
        sum = self.dfs(root.left, low, high, sum)
        sum = self.dfs(root.right, low, high, sum)
        return sum

File: 0938-Range-Sum-of-BST/0938-Range-Sum-of-BST.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int sum = 0;
    void dfs(TreeNode* root, int low, int high){
        if(root == NULL) return;
        //addition of nodes which are within given range
        if(root->val >= low && root->val <= high){
            sum += root->val;
        }
        // for left traversal
        if(root->val > low) dfs(root->left, low, high);
        // for right traversal
        if(root->val < high) dfs(root->right, low, high);
    }
    int rangeSumBST(TreeNode* root, int low, int high) {
        if(root == NULL) return 0;
        dfs(root, low, high);
        return sum;
    }
};
